src/cueUtilities.js

```diff
@@ -196,139 +220,193 @@ module.exports = function (params, cy, api) {
-		});
-
-		cy.on('grab', 'node', data.eGrab = function (e) {
-			preventDrawing = true;
-		});
-
-		cy.on('free', 'node', data.eFree = function (e) {
-			preventDrawing = false;
-		});
-
-		cy.on('position', 'node', data.ePosition = function () {
-			if (nodeWithRenderedCue)
-				clearDraws();
-		});
-
-		cy.on('remove', 'node', data.eRemove = function () {
-			clearDraws();
-			nodeWithRenderedCue = null;
-		});
-
-		var ur;
-		cy.on('select', 'node', data.eSelect = function(){
-			if (this.length > cy.nodes(":selected").length)
-				this.unselect();
-		});
-
-		cy.on('tap', 'node', data.eTap = function (event) {
-			var node = nodeWithRenderedCue;
-      var opts = options();
-			if (node){
-				var expandcollapseRenderedStartX = node._private.data.expandcollapseRenderedStartX;
-				var expandcollapseRenderedStartY = node._private.data.expandcollapseRenderedStartY;
-				var expandcollapseRenderedRectSize = node._private.data.expandcollapseRenderedCueSize;
-				var expandcollapseRenderedEndX = expandcollapseRenderedStartX + expandcollapseRenderedRectSize;
-				var expandcollapseRenderedEndY = expandcollapseRenderedStartY + expandcollapseRenderedRectSize;
-                
-                var cyRenderedPos = event.renderedPosition || event.cyRenderedPosition;
-				var cyRenderedPosX = cyRenderedPos.x;
-				var cyRenderedPosY = cyRenderedPos.y;
-				var factor = (opts.expandCollapseCueSensitivity - 1) / 2;
-
-				if ( (Math.abs(oldMousePos.x - currMousePos.x) < 5 && Math.abs(oldMousePos.y - currMousePos.y) < 5)
-					&& cyRenderedPosX >= expandcollapseRenderedStartX - expandcollapseRenderedRectSize * factor
-					&& cyRenderedPosX <= expandcollapseRenderedEndX + expandcollapseRenderedRectSize * factor
-					&& cyRenderedPosY >= expandcollapseRenderedStartY - expandcollapseRenderedRectSize * factor
-					&& cyRenderedPosY <= expandcollapseRenderedEndY + expandcollapseRenderedRectSize * factor) {
-					if(opts.undoable && !ur)
-						ur = cy.undoRedo({
-							defaultActions: false
-						});
-					if(api.isCollapsible(node))
-						if (opts.undoable){
-							ur.do("collapse", {
-								nodes: node,
-								options: opts
-							});
-						}
-						else
-							api.collapse(node, opts);
-				else if(api.isExpandable(node))
-					if (opts.undoable)
-						ur.do("expand", {
-							nodes: node,
-							options: opts
-						});
-					else
-						api.expand(node, opts);
-					}
-			}
-		});
+        function selectedGroupsContainsGroup(group) {
+          for (let i = 0; i < selectedGroups.length; i++) {
+            if (selectedGroups[i].id() === group.id()) {
+              return true;
+            }
+          }
+          return false;
+        }
+
+        cy.on('mouseout', 'node', data.eMouseOut = function(e) {
+          let node = this;
+          if (mouseIsHoveringOver(node)) {
+            clearDraws();
+            hoveredGroup = null;
+            drawCuesForSelectedGroups();
```

Comment:
Do you need to check `options().appearOnGroupSelect` here before drawing selected group cues?

Reply:
It is not needed. This works regardless of appearOnGroupSelect.

Reply:
What makes it work? Why do you need to check it elsewhere but not here?

Reply:
It's there because we may hover over a node inside the group, so it erases the cue. If we go back onto a group, it won't do anything because the mouseout node is not a group.